pkg/api/write.go

```diff
@@ -76,7 +78,7 @@ func Write(writer ingestor.DBInserter, elector *util.Elector, metrics *Metrics)
 		metrics.ReceivedSamples.Add(float64(receivedBatchCount))
 		begin := time.Now()
 
-		numSamples, err := writer.Ingest(timeseries, req)
+		numSamples, maxt, err := writer.Ingest(timeseries, req)
```

Comment:
Why is the maxt not the highest received timestamp? 

```
for _, si := range data {
	numRows += uint64(si.CountSamples())
	// Since by default, samples are expected to arrive in sorted order with time,
	// the last sample should be the maxt of that series. This assumption helps avoid costly inner loops.
	ls := si.LastSample()
	if maxt < ls.Timestamp {
		maxt = ls.Timestamp
	}
}
```

This is the code for maxt.

For the highest timestamp sent to db, we have `promscale_max_sent_timestamp_milliseconds`. That code lies in line 393 in `ingestor/insert.go`

Reply:
Ah got it. In that case, we should keep this (with the updates to names) and remove the ones in insert.go